presto-spi/src/main/java/io/prestosql/spi/connector/UpdateKind.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+public enum UpdateKind
+{
+    NO_UPDATE,
```

Comment:
Instead of representing "no update" a type of update, make callers use an `Optional<UpdateKind>`. This will also make callsites where an UpdateKind is needed more obvious and coherent. For instance, in:

```
public Scope analyzeForUpdate(Table table, Optional<Scope> outerQueryScope, UpdateKind updateKind)
```

It would be wrong or strange to call that method with `UpdateKind.NO_UPDATE` (and the method would need to check internally for this case).

Reply:
Converted to Optional<UpdateKind>, a much cleaner solution, making the NO_UPDATE enum value vanish.